### ISSUE TITLE
version_string is not defined

### ISSUE THREAD

**Issue by pllim** [spacetelescope/asdf#209]
As reported by @hbushouse:

_I, along with at least 1 other user in INS, are running into the following error when attempting to run anything from the jwst package using either the astroconda or jwstb6 environments. I’m running from my Mac laptop (don’t know what the INS user is on). Does anyone have any clue as to what the problem is?_

```
Traceback (most recent call last):
  File "/Users/bushouse/JWST/softw/jwst/scripts/strun", line 14, in <module>
    step = Step.from_cmdline(sys.argv[1:])
  File "/Users/bushouse/JWST/softw/jwst/jwst/stpipe/step.py", line 128, in from_cmdline
    return cmdline.step_from_cmdline(args)
  File "/Users/bushouse/JWST/softw/jwst/jwst/stpipe/cmdline.py", line 269, in step_from_cmdline
    step.run(*positional)
  File "/Users/bushouse/JWST/softw/jwst/jwst/stpipe/step.py", line 322, in run
    self._precache_reference_files(args[0])
  File "/Users/bushouse/JWST/softw/jwst/jwst/stpipe/step.py", line 458, in _precache_reference_files
    model = datamodels.open(input_file)
  File "/Users/bushouse/JWST/softw/jwst/jwst/datamodels/__init__.py", line 195, in open
    return new_class(init, extensions=extensions)
  File "/Users/bushouse/JWST/softw/jwst/jwst/datamodels/image.py", line 34, in __init__
    super(ImageModel, self).__init__(init=init, **kwargs)
  File "/Users/bushouse/JWST/softw/jwst/jwst/datamodels/model_base.py", line 156, in __init__
    pass_invalid_values=self._pass_invalid_values)
  File "/Users/bushouse/JWST/softw/jwst/jwst/datamodels/fits_support.py", line 565, in from_fits
    ff = fits_embed.AsdfInFits.open(hdulist, extensions=extensions)
  File "//anaconda/lib/python2.7/site-packages/asdf/fits_embed.py", line 160, in open
    self = cls(hdulist, uri=uri, extensions=extensions)
  File "//anaconda/lib/python2.7/site-packages/asdf/fits_embed.py", line 146, in __init__
    tree=tree, uri=uri, extensions=extensions)
  File "//anaconda/lib/python2.7/site-packages/asdf/asdf.py", line 87, in __init__
    self.tree = {}
  File "//anaconda/lib/python2.7/site-packages/asdf/asdf.py", line 248, in tree
    self._validate(asdf_object)
  File "//anaconda/lib/python2.7/site-packages/asdf/asdf.py", line 260, in _validate
    tree, self)
  File "//anaconda/lib/python2.7/site-packages/asdf/yamlutil.py", line 238, in custom_tree_to_tagged_tree
    return treeutil.walk_and_modify(tree, walker)
  File "//anaconda/lib/python2.7/site-packages/asdf/treeutil.py", line 188, in walk_and_modify
    return recurse(top)
  File "//anaconda/lib/python2.7/site-packages/asdf/treeutil.py", line 149, in recurse
    result = callback(result)
  File "//anaconda/lib/python2.7/site-packages/asdf/yamlutil.py", line 233, in walker
    tag = ctx.type_index.from_custom_type(type(node), ctx.version_string)
  File "//anaconda/lib/python2.7/site-packages/asdf/asdftypes.py", line 207, in from_custom_type
    write_type_index = _AsdfWriteTypeIndex(version, self)
  File "//anaconda/lib/python2.7/site-packages/asdf/asdftypes.py", line 105, in __init__
    version_map = versioning.get_version_map(version)
  File "//anaconda/lib/python2.7/site-packages/asdf/versioning.py", line 40, in get_version_map
    "Could not load version map for version {0}".format(version_string))
NameError: global name 'version_string' is not defined
```

**Comment by hbushouse** [comment]
But there seems to be more to it than that. The version_string is being used in the error statement in the except part of a try/except block, so in addition to this object not being defined, the real problem is why it hit that exception to begin with, which seems to be due to the fact that it couldn't find or load the asdf version_map that it's looking for.

**Comment by bernie-simon** [comment]
On Jul 7, 2016, at 1:57 PM, Howard Bushouse <notifications@github.com> wrote:

> But there seems to be more to it than that. The version_string is being used in the error statement in the except part of a try/except block, so in addition to this object not being defined, the real problem is why it hit that exception to begin with, which seems to be due to the fact that it couldn't find or load the asdf version_map that it's looking for.

It�s a pretty good bet that the asdf-standard directory was not installed. It�s a subpackage (or whatever git calls that) and needs to be installed separately from asdf.

**Comment by bernie-simon** [comment]
On Jul 7, 2016, at 2:03 PM, Bernie Simon <bsimon@stsci.edu> wrote:

> 
> On Jul 7, 2016, at 1:57 PM, Howard Bushouse <notifications@github.com> wrote:
> 
>> But there seems to be more to it than that. The version_string is being used in the error statement in the except part of a try/except block, so in addition to this object not being defined, the real problem is why it hit that exception to begin with, which seems to be due to the fact that it couldn't find or load the asdf version_map that it's looking for.
> 
> It�s a pretty good bet that the asdf-standard directory was not installed. It�s a subpackage (or whatever git calls that) and needs to be installed separately from asdf.

Search for version_map-1.0.0.yaml on the machine you�re running the jwst pipeline on. Should find it under asdf/asdf-standard/schemas. If not, something was not installed correctly.

find . -name version_map-1.0.0.yaml

**Comment by hbushouse** [comment]
On my system the get_version_map function is constructing a version_map_path of 'file://anaconda/envs/jwstb6/lib/python2.7/site-packages/asdf/schemas/stsci.edu/asdf/version_map-1.0.0.yaml', which fails in the call to generic_io.get_file(version_map_path). If I strip off the first 6 characters of that version_map_path (i.e. the file:/ part), it succeeds in opening the file.

**Comment by bernie-simon** [comment]
On Jul 7, 2016, at 4:18 PM, Howard Bushouse <notifications@github.com> wrote:

> On my system the get_version_map function is constructing a version_map_path of 'file://anaconda/envs/jwstb6/lib/python2.7/site-packages/asdf/schemas/stsci.edu/asdf/version_map-1.0.0.yaml', which fails in the call to generic_io.get_file(version_map_path). If I strip off the first 6 characters of that version_map_path (i.e. the file:/ part), it succeeds in opening the file.


Well, it�s a little complicated. your version_map_path gets fed into resolver.DEFAULT_URL_MAPPING to return a file name. So if DEFAULT_URL_MAPPING is not defined correctly, you�re not going to get the proper file name and not going to find the file. DEFAULT_URL_MAPPING is defined at the bottom of resolver.py and it, in turn is defined by SCHEMA_PATH, which is at the top of resolver.py and STSCI_SCHEMA_URI_BASE, which is defined in constants,py. If something is wrong in that chain, you get the error message.

**Comment by nden** [comment]
@hbushouse @pllim @bernie-simon Do you see any more problems like this? Can this be closed?
The fix for the immediate error was implemented in #210.

**Comment by pllim** [comment]
I don't really use this but I notice that PR is not merged yet.

**Comment by nden** [comment]
Fixed in #217
